tasks/post_to_database/post_to_database.py

```diff
@@ -0,0 +1,249 @@
+
+    Args:
+        record: Contains the following keys:
+            'body' (str): A json string representing a dict.
+                Contains key/value pairs of column names and values for those columns.
+            'messageAttributes' (dict): Contains the following keys:
+                'TableName' (str): The name of the table to target.
+                'RequestMethod' (str): 'post' or 'put', depending on if row should be created or updated respectively.
+        db_connect_info: See shared_db.py's get_configuration for further details.
+    """
+    values = json.loads(record['body'])
+    request_method = RequestMethod(record['messageAttributes']['RequestMethod'])
+    if request_method == RequestMethod.NEW_JOB:
+        new_job_validate(values)
+        create_status_for_job_and_files(values['job_id'],
+                                        values['granule_id'],
+                                        values['request_time'],
+                                        values['archive_destination'],
+                                        values['files'],
+                                        db_connect_info)
+    elif request_method == RequestMethod.UPDATE_FILE:
+        update_file_validate(values)
+        update_status_for_file(values['job_id'],
+                               values['granule_id'],
+                               values['filename'],
+                               values['last_update'],
+                               values.get('completion_time', None),
+                               values['status_id'],
+                               values.get('error_message', None),
+                               db_connect_info)
+    else:
+        raise ValueError(f"RequestMethod '{request_method}' not found.")
+
+
+def create_status_for_job_and_files(job_id: str,
+                                    granule_id: str,
+                                    request_time: str,
+                                    archive_destination: str,
+                                    files: List[Dict[str, Any]],
+                                    db_connect_info: Dict) -> None:
+    """
+    Posts the entry for the job, followed by individual entries for each file.
+
+    Args:
+        job_id: The unique identifier used for tracking requests.
+        granule_id: The id of the granule being restored.
+        archive_destination: The S3 bucket destination of where the data is archived.
+        request_time: The time the restore was requested in utc and iso-format.
+        files: A List of Dicts with the following keys:
+            'filename' (str)
+            'key_path' (str)
+            'restore_destination' (str)
+            'status_id' (int)
+            'error_message' (str, Optional)
+            'request_time' (str)
+            'last_update' (str)
+            'completion_time' (str, Optional)
+        db_connect_info: See shared_db.py's get_configuration for further details.
+    """
+    found_pending = False
```

Comment:
Wondering if it would make sense to move this logic for setting status and completion time to the shared library so that it is put together with the proper status and times before arrival at this lambda. Thoughts?

Reply:
Makes sense. Changed.

Reply:
Having it here makes it harder to send junk data.

Reply:
I'd be open to having checks here and in the shared lib if that sounds good.